done/done.py

```diff
@@ -1,77 +1,110 @@
-        return data.decode("utf8")
-
+    # pylint: disable=unused-argument
     @XBlock.json_handler
     def toggle_button(self, data, suffix=''):
-        self.done = data['done']
-        if data['done']:
-            grade = 1
-        else:
-            grade = 0
-
-        self.runtime.publish(self, 'grade', {'value':grade, 'max_value': 1})
-        return {}
-
-
-    def student_view(self, context=None):
+        """
+        Ajax call when the button is clicked. Input is a JSON dictionary
+        with one boolean field: `done`. This will save this in the
+        XBlock field, and then issue an appropriate grade.
+        """
+        if 'done' in data:
+            self.done = data['done']
+            if data['done']:
+                grade = 1
+            else:
+                grade = 0
+            grade_event = {'value': grade, 'max_value': 1}
+            self.runtime.publish(self, 'grade', grade_event)
+            # This should move to self.runtime.publish, once that pipeline
+            # is finished for XBlocks.
+            tracker.emit("edx.done.toggle", {'done': self.done})        
+
+        return {'state': self.done}
+
+    def student_view(self, context=None):  # pylint: disable=unused-argument
         """
         The primary view of the DoneXBlock, shown to students
         when viewing courses.
         """
-        html = self.resource_string("static/html/done.html")
-        frag = Fragment(html)#.format(uid=self.scope_ids.usage_id))
-        frag.add_css_url("//ajax.googleapis.com/ajax/libs/jqueryui/1.10.4/themes/smoothness/jquery-ui.css")
-        #frag.add_javascript_url("//ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js")
-        frag.add_javascript_url("//ajax.googleapis.com/ajax/libs/jqueryui/1.10.4/jquery-ui.min.js")
-        frag.add_css(self.resource_string("static/css/done.css"))
-        grow_left = 1
-        grow_right = 1
-        if self.align.lower() == "left":
-            grow_left = 0
-        if self.align.lower() == "right":
-            grow_right = 0
-        frag.add_css(".done_left_spacer {{ flex-grow:{l}; }} .done_right_spacer {{ flex-grow:{r}; }}".format(r=grow_right, l=grow_left))
-        frag.add_javascript(self.resource_string("static/js/src/done.js"))
-        if self.done:
-            frag.initialize_js("DoneXBlockOn")
-        else:
-            frag.initialize_js("DoneXBlockOff")
+        html_resource = resource_string("static/html/done.html")
+        html = html_resource.format(done=self.done,
+                                    id=uuid.uuid1(0))
+        (unchecked_png, checked_png) = (
+            self.runtime.local_resource_url(self, x) for x in
+            ('public/check-empty.png', 'public/check-full.png')
+        )
+
+        frag = Fragment(html)
+        frag.add_css(resource_string("static/css/done.css"))
+        frag.add_javascript(resource_string("static/js/src/done.js"))
+        frag.initialize_js("DoneXBlock", {'state': self.done,
```

Comment:
If the name of the xblock is DoneXBlock, why is "done" what is specified in Advanced Settings to add the xblock? I'm just getting confused about the name because there seem to be a variety of names-- "DoneXBlock", "done", "Completion".